src/components/Menu/Menu.tsx

```diff
@@ -0,0 +1,116 @@
+/** @jsx jsx */
+import { jsx, css } from '@emotion/core';
+import React from 'react';
+// import { buttonSpanStyle, menuStyle } from './Menu.style';
+import useTheme from 'hooks/useTheme';
+import Button from 'components/Button';
+import Icon from '../Icon';
+import { EventProps } from 'utils/common';
+import ClickAwayListener from 'components/utils/ClickAwayListener';
+import { rem, darken } from 'polished';
+import { AcceptedColorComponentTypes } from 'utils/themeFunctions';
+
+export type Props = {
+  /** Items that are being declared as menu options */
+  items: string[];
+  /** Returns the items selected on the menu */
+  selectedItem: string | null;
+  /** A callback that is being triggered when an items has been clicked */
+  onSelect: (option: string) => void;
+  /** The text of the button to show - defaults to "More" */
+  buttonText: React.ReactNode;
+  /** Menu position when open */
+  menuPosition?: 'left' | 'right';
+  /** Indicator to show dots icon */
+  showOptionIcon?: boolean;
+  /** The type of the button - defaults to "primary" */
+  buttonType?: AcceptedColorComponentTypes;
+};
+
+export type TestProps = {
+  dataTestId?: string;
+};
+
+const Menu: React.FC<Props & TestProps & EventProps> = props => {
+  const {
+    items,
+    onSelect,
+    buttonText = 'More',
+    menuPosition = 'left',
+    showOptionIcon = false,
+    buttonType = 'primary',
+  } = props;
+  const [open, setOpen] = React.useState(false);
+  const theme = useTheme();
+
+  return (
+    <ClickAwayListener onClick={() => setOpen(false)}>
+      <div css={{ position: 'relative', display: 'inline-block' }}>
+        <Button
+          onClick={() => setOpen(!open)}
+          type={buttonType}
+          iconAlign={showOptionIcon ? 'right' : undefined}
+          icon={
+            showOptionIcon ? (
+              <Icon
+                name={'dotsVertical'}
+                color={buttonType === ('primary' || 'secondary') ? 'dark' : 'light'}
+              />
+            ) : null
+          }
+        >
+          <span>{buttonText}</span>
+        </Button>
+        {open && (
+          <div
+            css={css`
+              position: absolute;
+              top: ${rem(48)};
+              left: ${menuPosition === 'left' ? 0 : 'initial'};
+              right 0;
+              width: ${rem(148)};
+              height: auto;
+              background-color: #fff;
+              box-shadow: 0px 0px ${rem(16)} grey;
+              display: flex;
+              flex-direction: column;
+              justify-content: space-evenly;
+              text-align: center;
+              border-radius: ${rem(4)};
+              z-index: 1;
```

Comment:
Maybe this should be initialized from the theme so it can be configured according to each project's needs and special cases.

Reply:
indexing ?

Reply:
yes

Reply:
you could make the parent higher :) rather than changing the component

Reply:
But you would have to add an extra layer for this. 

I like the way material allows full customization over all of the components they have. 

If we are to leave it like this and not have some kind of manipulating/changing some values we should ask for the hierarchy of the components according to the importance they have and style them accordingly. So at least, when they are on the same level, they will display properly with no overlaps. 

(This is not something to be implemented on this specific PR. Maybe discuss this further on an issue?)

Reply:
i see the importance in this and i created a new issue for it on #35 for further discussion and you can put your opinion or thoughts or even documentation in there